File: is-subsequence/is-subsequence.py
class Solution(object):
    def isSubsequence(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        if (s == t) or (s == ''):
            return True

        elif len(s) == 1:
            if s in t:
                return True
            return False

        else:
            sIndex = 0
            for tIndex in range(len(t)):
                print(sIndex)
                tChar = t[tIndex]
                sChar = s[sIndex]
                if sChar == tChar:
                    sIndex += 1
                if sIndex == len(s):
                    return True

            return False

File: easy/twoPointers/isSubsequence.ts
export function isSubsequence(s: string, t: string): boolean {
  let sIndex: number = 0;

  for (const char of t) {
    if (char === s[sIndex]) {
      sIndex++;
    }
  }

  return sIndex === s.length;
}

export function isSubsequenceFast(s: string, t: string): boolean {
  let i = 0,
    j = 0;

  while (i < s.length && j < t.length) {
    if (s[i] === t[j]) i++;
    j++;
  }

  return i === s.length;
}
